From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
Date: Tue, 29 Oct 2024 19:51:45 +0530
Subject: drm/xe: control dma_fence_array_alloc unavailability

dma_fence_array_create was converted to two different functions
dma_fence_array_alloc and dma_fence_array_init in KV6.12. use older API
for older versions.

Error:
--------------------------------------------------------------------------
error: implicit declaration of function _dma_fence_array_alloc
error: implicit declaration of function _dma_fence_array_init
--------------------------------------------------------------------------

Reference-Id:
ddc94d0b17e8
dma-buf: Split out dma fence array create into alloc and arm functions

Signed-off-by: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
---
 drivers/gpu/drm/xe/xe_pt.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/gpu/drm/xe/xe_pt.c b/drivers/gpu/drm/xe/xe_pt.c
index d6353e8969f0..fe57c9b577d9 100644
--- a/drivers/gpu/drm/xe/xe_pt.c
+++ b/drivers/gpu/drm/xe/xe_pt.c
@@ -2031,11 +2031,13 @@ xe_pt_update_ops_run(struct xe_tile *tile, struct xe_vma_ops *vops)
 				err = -ENOMEM;
 				goto free_ifence;
 			}
+#ifndef BPM_DMA_FENCE_ARRAY_ALLOC_NOT_PRESENT
 			cf = dma_fence_array_alloc(2);
 			if (!cf) {
 				err = -ENOMEM;
 				goto free_ifence;
 			}
+#endif
 		}
 	}
 
@@ -2079,10 +2081,21 @@ xe_pt_update_ops_run(struct xe_tile *tile, struct xe_vma_ops *vops)
 						pt_update_ops->last, vm->usm.asid);
 			fences[0] = &ifence->base.base;
 			fences[1] = &mfence->base.base;
+#ifdef BPM_DMA_FENCE_ARRAY_ALLOC_NOT_PRESENT
+			cf=dma_fence_array_create(2, fences,
+						vm->composite_fence_ctx,
+						vm->composite_fence_seqno++,
+						false);
+			if (!cf) {
+				err = -ENOMEM;
+				goto free_fence;
+			}
+#else
 			dma_fence_array_init(cf, 2, fences,
 					     vm->composite_fence_ctx,
 					     vm->composite_fence_seqno++,
 					     false);
+#endif
 			fence = &cf->base;
 		} else {
 			fence = &ifence->base.base;
@@ -2118,10 +2131,16 @@ xe_pt_update_ops_run(struct xe_tile *tile, struct xe_vma_ops *vops)
 
 	return fence;
 
+#ifdef BPM_DMA_FENCE_ARRAY_ALLOC_NOT_PRESENT
+free_fence:
+	kfree(fence);
+#endif
 free_rfence:
 	kfree(rfence);
 free_ifence:
+#ifndef BPM_DMA_FENCE_ARRAY_ALLOC_NOT_PRESENT
 	kfree(cf);
+#endif
 	kfree(fences);
 	kfree(mfence);
 	kfree(ifence);
-- 
2.25.1

